src/test/regress/expected/merge_repartition.out

```diff
@@ -999,6 +999,165 @@ SQL function "compare_data" statement 2
 
 (1 row)
 
+--
+-- Test target with false clause
+--
+SELECT cleanup_data();
+NOTICE:  creating a new table for merge_repartition_schema.citus_target
+CONTEXT:  SQL function "cleanup_data" statement 5
+NOTICE:  moving the data of merge_repartition_schema.citus_target
+CONTEXT:  SQL function "cleanup_data" statement 5
+NOTICE:  dropping the old merge_repartition_schema.citus_target
+CONTEXT:  SQL function "cleanup_data" statement 5
+NOTICE:  renaming the new table to merge_repartition_schema.citus_target
+CONTEXT:  SQL function "cleanup_data" statement 5
+NOTICE:  creating a new table for merge_repartition_schema.citus_source
+CONTEXT:  SQL function "cleanup_data" statement 6
+NOTICE:  moving the data of merge_repartition_schema.citus_source
+CONTEXT:  SQL function "cleanup_data" statement 6
+NOTICE:  dropping the old merge_repartition_schema.citus_source
+CONTEXT:  SQL function "cleanup_data" statement 6
+NOTICE:  renaming the new table to merge_repartition_schema.citus_source
+CONTEXT:  SQL function "cleanup_data" statement 6
+ cleanup_data
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT setup_data();
+ setup_data
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT create_distributed_table('citus_target', 'id');
+NOTICE:  Copying data from local table...
+NOTICE:  copying the data has completed
+DETAIL:  The local data in the table is no longer visible, but is still on disk.
+HINT:  To remove the local data, run: SELECT truncate_local_data_after_distributing_table($$merge_repartition_schema.citus_target$$)
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT create_distributed_table('citus_source', 'id');
+NOTICE:  Copying data from local table...
+NOTICE:  copying the data has completed
+DETAIL:  The local data in the table is no longer visible, but is still on disk.
+HINT:  To remove the local data, run: SELECT truncate_local_data_after_distributing_table($$merge_repartition_schema.citus_source$$)
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+MERGE INTO pg_target t
+USING (SELECT * FROM pg_source WHERE id > 25000) AS s
+ON t.id = s.id AND t.id < 25000
+WHEN MATCHED AND t.id <= 55000 THEN
```

Comment:
Historically, we had many edge cases/bugs with `false` clauses. Apparently many ORMs injects `WHERE false` to somewhere in the query  when they want the operation to be a no-op -- instead of skipping the operation.

So, maybe add  explicit tests with 

-WHEN MATCH and FALSE 
-WHEN NOT MATCH and 1=0 
 -ON t.key = s.key AND 1 >0
 -(SELECT * FROM source WHERE false) as source

Reply:
@onderkalaci  you meant

-ON t.key = s.key AND 1 >0 
to
-ON t.key = s.key AND 1 < 0

Reply:
Added unit tests.